### ISSUE TITLE
Migrate to ginkgo v2

### ISSUE THREAD

**Issue by jakobmoellersap** [kubernetes-sigs/kubebuilder#2875]
### What do you want to happen?

With the Migration of Gingko in controller-runtime (see https://github.com/kubernetes-sigs/controller-runtime/pull/1977) we will also have the chance to migrate our test coding as well as our templated scaffolds to gingko v2. To make sure that we do not break incompatibility, we especially need to look at the migration guide in https://onsi.github.io/ginkgo/MIGRATING_TO_V2

~Since CR is upgrading this also means we have to react to the resulting breaking changes~
EDIT: This is not a breaking change anymore after the removal of the changes of the printer package

> The PR removes two helpers in pkg/envtest/printer that were used with ginkgo v1. Since you can't use both versions at the same time, this isn't relevant though.
In summary, all projects depending on controller-runtime will have to update to ginkgo v2 after this is merged and released. Upstream did so already: https://github.com/kubernetes/kubernetes/pull/109111

**Comment by jakobmoellersap** [comment]
Duplicate of https://github.com/kubernetes-sigs/kubebuilder/issues/2532 @camilamacedo86 which one should we leave open?

**Comment by camilamacedo86** [comment]
We can close this one in favour of https://github.com/kubernetes-sigs/kubebuilder/issues/2532
@jakobmoellersap also I think we can check what also needs to be done in https://github.com/kubernetes-sigs/kubebuilder/pull/2522 and would be nice to add this contributor as co-author to also give the credit to him. 

Closing this one and adding the info on why we will move forward with it for v3 and project as well there.
